src/player.cpp

```diff
@@ -9059,12 +9314,10 @@ void player::do_read( item &book )
                 }
             }
 
-            if( skill_level == reading->level || !skill_level.can_train() ) {
-                if( learner->is_player() ) {
-                    add_msg( m_info, _( "You can no longer learn from %s." ), book.type_name().c_str() );
-                } else {
-                    cant_learn.insert( learner->disp_name() );
-                }
+            if( ( skill_level == reading->level || !skill_level.can_train() ) || 
+              ( ( g->u.has_trait( trait_id( "SCHIZOPHRENIC" ) ) || g->u.has_artifact_with( AEP_SCHIZO ) ) &&
```

Comment:
This looks strange, why is this suddenly refering to `g->u` when there are several `player` objects used in this function (`this`, `learner` to name some). It should probably be `learner` instead of `g->u`.